### ISSUE TITLE
Update outdated "Code Structure" wiki page

### ISSUE THREAD

**Issue by greene858** [department-of-veterans-affairs/abd-vro#1109]
**User Story**

As a VRO engineer I would like an overview of the code structure of the project, so that I can quickly examine the project, navigate to key sections, and understand the organization and extent of the project.

- Update Code structure page
- Validate that all sections are acurate to current code base
- add about how the annotations from IBM/MAS should be interpreted. Rajesh showed a Word document that should be migrated (and updated) to the wiki.

**Acceptance Criteria**

1. All lists of components acurately reflect iMVP code base initial release
2. Diagrams acurately reflect iMVP code base initial release
3. All links connect to valid pages
4. IBM/MAS annotations are explained
5. Any deprecated elements are removed

**Comment by yoomlam** [comment]
The desired code structure is documented in [Software Conventions](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Software-Conventions).
PR #1330 is a first step to get it closer to the convention.

**Comment by msnwatson** [comment]
Okay, I've taken a look at the word documents mentioned in the issue description here for getting more information about the MasAnnotation objects. This seems to be related to the API response from MAS, and so it seems more intuitive to me to have this information in the code. I drafted https://github.com/department-of-veterans-affairs/abd-vro/pull/1475 to add some comments to aid in interpretation of what the MasAnnotation object is used for. I didn't see any information about IBM.

Bigger question though is why the models for the data returned by MAS api are not stored in the `api` subproject but rather in a subdirectory of the `shared` subproject. 

cc @yoomlam

**Comment by yoomlam** [comment]
Thanks for the PR!

Note that the `api` subproject will be removed after #1330 merges. According to the current Software Conventions, models referenced in VRO's API should be under the respective domain folder. Specifically, `MasAnnotation` should be under the `domain-rrd` folder.

**Comment by msnwatson** [comment]
Once the PR merges, the MAS annotations should be explained.

I also made the following revisions or new pages to the wiki:

- https://github.com/department-of-veterans-affairs/abd-vro/wiki/Local-Setup
- https://github.com/department-of-veterans-affairs/abd-vro/wiki/Code-structure/_compare/49aed54a95b78451feb520dd3a84124195a18c81
- https://github.com/department-of-veterans-affairs/abd-vro/wiki/Code-structure/_compare/0b838e64b89ae76d84abd569cc16bdc49a44b851
- https://github.com/department-of-veterans-affairs/abd-vro/wiki/Gradle/_compare/3ea9b3af3b94d546d0aa8728e57829b747711952

If I'm understanding correctly the Software Conventions page was created after the Code Structure page, and I think I was a bit confused on what the distinct goals of the two pages were. I tried to emphasize that the Code Structure page is a description of the mechanisms by which the conventions are implemented within the codebase. I'm afraid to go into to much depth of the different components or microservices right now as it seems like the structure could change significantly with the handoff and refactoring.
